modules/core/src/main/scala/com.snowplowanalytics.iglu/client/validator/CirceValidator.scala

```diff
@@ -102,4 +106,112 @@ object CirceValidator extends Validator[Json] {
 
   private def fromValidationMessage(m: ValidationMessage): ValidatorReport =
     ValidatorReport(m.getMessage, m.getPath.some, m.getArguments.toList, m.getType.some)
+
+  private def evaluateSchema(
+    schemaAsNode: JsonNode
+  ): Either[ValidatorError.InvalidSchema, JsonSchema] = {
+    Either
+      .catchNonFatal(
+        IgluMetaschemaFactory
+          .getSchema(schemaAsNode, SchemaValidatorsConfig)
+      )
+      .leftMap(ValidatorError.schemaIssue)
+  }
+
+  private def validateSchemaAgainstV4(schema: JsonNode): List[ValidatorError.SchemaIssue] = {
+    V4Schema
+      .validate(schema)
+      .asScala
+      .toList
+      .map(m => ValidatorError.SchemaIssue(m.getPath, m.getMessage))
+  }
+
+  private[client] object WithCaching {
```

Comment:
This implementation is prone to race conditions, as LruMap updates are not atomic and need to be protected by the Semaphor. 
Check out Chris Davenport's version: [here](https://gist.github.com/ChristopherDavenport/8364fa2ca14557fce12baca6cb38d5b8#file-lrucache-scala)
 
However, those are rare and occasional item deletion is not too important in the context of this library.